src/AdminCli/Infrastructure/CommandLineApp.cs

```diff
@@ -0,0 +1,32 @@
+using AdminCli.CliCommands;
+using McMaster.Extensions.CommandLineUtils;
+using Microsoft.Extensions.DependencyInjection;
+
+namespace AdminCli.Infrastructure;
+
+public static class CommandLineApp
+{
+    public static IServiceCollection AddCommandLineApp(this IServiceCollection services)
+    {
+        return services.AddSingleton(container =>
+        {
+            var app = new CommandLineApplication
+            {
+                Name = "admin-cli",
+                Description = "The Command Line Interface of the Thinktecture Cloud Native Sample that allows admins to run internal commands."
+            };
+            app.HelpOption(inherited: true);
+
+            var context = new CommandConfigurationContext(app, container);
```

Comment:
Leveraging the Attributes provided by `McMaster.Extensions.CommandLineUtils` would have made the entire CLI application way more simple and easier to grasp for other devs. 

Not saying or judging about the chosen path, but It would have been faster and easier (from implementation point of view)  😆

Reply:
I choose this path because debugging is easier.